lib/OpenQA/Git.pm

```diff
@@ -21,16 +22,16 @@ sub config ($self, $args = undef) {
     return $app->config->{'scm git'};
 }
 
-sub _prepare_git_command ($self, $args = undef) {
-    my $dir = $args->{dir} // $self->dir;
+sub _prepare_git_command ($self) {
+    my $dir = $self->dir;
     if ($dir !~ /^\//) {
         my $absolute_path = abs_path($dir);
         $dir = $absolute_path if ($absolute_path);
     }
     return ('git', '-C', $dir);
 }
 
-sub _format_git_error ($result, $error_message) {
+sub _format_git_error ($self, $result, $error_message) {
     if ($result->{stderr} or $result->{stdout}) {
         $error_message .= ': ' . $result->{stdout} . $result->{stderr};
     }
```

Comment:
Why don't we have the path in the error message anymore now?
We did this together, and I think if we get an error it would be good to know which path it was happening in.
The original code has the path in the error message.

Reply:
Ah, i misunderstood something before, in [16-utils-runcmd.t#L51](https://github.com/r-richardson/openQA/blob/301acc6ceb68654f17caee1de9438ad318504762/t/16-utils-runcmd.t#L51), can i just replace the `tempdir()` call with e.g. `./temp-dir` ? Because currently the paths contain random strings which make it impossible to prepare the string for comparison in the tests. Or should i only display the path conditionally instead ?

```
t/16-utils-runcmd.t .. 1/? 
    #   Failed test 'Git error message returned'
    #   at t/16-utils-runcmd.t line 56.
    #                   'Unable to commit via Git /tmp/_gssEGO8wE: fatal: not a git repository (or any parent up to mount point /)
    # Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).'
    #     doesn't match '(?^u:^Unable to commit via Git: fatal: (N|n)ot a git repository \(or any)'
    # Looks like you failed 1 test of 3.
```

Reply:
You're right. I misunderstood the way the git_format output is tested and thought it does not make sense here. Dir output is back now and tests adjusted